src/buildercore/threadbare/operations.py

```diff
@@ -0,0 +1,638 @@
+import tempfile
+import contextlib
+import subprocess
+from threading import Timer
```

Comment:
:policeman: mixing processes and threads - will get to the point where this is used

Reply:
Threadbare has a well defined scope and an established project (Fabric) to take lessons from. I have no intention of swapping multiprocessing for threading.

That said, you shouldn't be afraid of threading but of programming languages and paradigms that let you shoot yourself in the foot with threads.